File: read_excel.py
import pandas as pd
import json
import traceback

try:
    file_path = "C:/Users/prems/Downloads/Personal Pipeline (1).xlsx"
    xl = pd.ExcelFile(file_path)
    result = {}
    for sheet_name in xl.sheet_names:
        df = xl.parse(sheet_name)
        df = df.where(pd.notnull(df), None) # Replace NaN with None
        result[sheet_name] = df.to_dict(orient='records')
    
    with open('excel_dump.json', 'w', encoding='utf-8') as f:
        json.dump(result, f, indent=2, default=str)
    print("Success")
except Exception as e:
    print("Error:", e)
    traceback.print_exc()

File: read_excel.mjs
import * as xlsx from 'xlsx';
import * as fs from 'fs';

try {
    const workbook = xlsx.readFile('C:/Users/prems/Downloads/Personal Pipeline (1).xlsx');
    const result = {};

    workbook.SheetNames.forEach(sheetName => {
        const sheet = workbook.Sheets[sheetName];
        result[sheetName] = xlsx.utils.sheet_to_json(sheet, { defval: null });
    });

    fs.writeFileSync('excel_dump.json', JSON.stringify(result, null, 2));
    console.log('Successfully extracted Excel data to excel_dump.json');
} catch (e) {
    console.error('Error reading Excel file:', e);
}
